### ISSUE TITLE
a shared state store?

### ISSUE THREAD

**Issue by shapiromatron** [kkinder/puepy#13]
This project looks awesome, I'm excited to try it out on a few projects we're working on.

One question I had though after reading through the docs. A common pattern we use in our web apps is a single state store, and then multiple components can read/write to the state store, with components being updated reactively. In JS, we use [mobx](https://mobx.js.org/), though there other other libraries that accomplish something similar. 

Is this pattern possible with puepy? It's seems better than passing state up/down the component tree. I didn't see any examples that explored this concept, though maybe I was missing it.

**Comment by kkinder** [comment]
There is, sort of.

You can set state on the application. The tutorial's "full app" demo makes use of this, though it could be documented better. Take a look at the running demo here:

https://kkinder.pyscriptapps.com/puepy-tutorial/latest/tutorial/10_full_app/index.html#/login

If you look at the [source code in PySript.com](https://pyscript.com/@kkinder/puepy-tutorial/latest), you'll notice it does this to log the user in:

```python
self.application.state["authenticated_user"] = self.state["username"]
```

And to log the user out:

```python
self.application.state["authenticated_user"] = ""
```

`self.application.state` is a reactive dictionary, just like component state. This should probably be documented better though.

Is that what you were looking for?

**Comment by shapiromatron** [comment]
Yes, exactly what I was looking for. Thank you! Great project, looking forward to to trying this out.
